src/D2L.CodeStyle.UnsafeStaticCounter/AnalyzedResults.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections.Generic;
+
+namespace D2L.CodeStyle.UnsafeStaticCounter {
+
+	internal sealed class AnalyzedResults {
+		public readonly int UnsafeStaticsCount;
+		public readonly int UnsafeNonReadonlyStaticsCount;
+		public readonly IEnumerable<AnalyzedType> UnsafeStaticsPerType;
+		public readonly IEnumerable<AnalyzedProject> UnsafeStaticsPerProject;
+		public readonly IEnumerable<AnalyzedStatic> RawResults;
+
+		public AnalyzedResults( AnalyzedStatic[] rawResults ) {
+			RawResults = rawResults;
+
+			UnsafeStaticsCount = rawResults.Length;
+
+			var unsafeStaticsPerProject = new Dictionary<string, AnalyzedProject>();
+			var unsafeStaticsPerType = new Dictionary<string, AnalyzedType>();
+
+			foreach( var result in rawResults ) {
+
+				// increment per project
+				var project = unsafeStaticsPerProject.GetOrAdd(
+						result.ProjectName,
+						() => new AnalyzedProject( result.ProjectName )
+					);
+				project.UnsafeStaticsCount++;
+
+				// increment per-type
+				if( result.FieldOrPropType != null ) {
+
+					var analyzedType = unsafeStaticsPerType.GetOrAdd(
+							result.FieldOrPropType,
+							() => new AnalyzedType( result.FieldOrPropType )
+						);
+					analyzedType.UnsafeStaticsCount++;
+				}
+			}
+
+			// move `it` to readonly count
+			AnalyzedType itType;
+			if( unsafeStaticsPerType.TryGetValue( "it", out itType ) ) {

```

Comment:
bug: the `it` type actually gets cleared here: https://github.com/Brightspace/D2L.CodeStyle/pull/22/files#diff-881dec3d46bb2d41a3afdd74af7aadf6R23
I should be looking at `result.Cause` instead. I'll fix this tomorrow.